WordPressAuthenticator/NUX/NUXButton.swift

```diff
@@ -1,54 +1,285 @@
+            normalImage = UIImage.renderBackgroundImage(fill: primaryNormalBackgroundColor, border: primaryNormalBorderColor, shadowOffset: Metrics.backgroundShadowOffset)
+            highlightedImage = UIImage.renderBackgroundImage(fill: primaryHighlightBackgroundColor, border: primaryHighlightBorderColor, shadowOffset: Metrics.backgroundShadowOffset)
+        } else {
+            normalImage = UIImage.renderBackgroundImage(fill: secondaryNormalBackgroundColor, border: secondaryNormalBorderColor, shadowOffset: Metrics.backgroundShadowOffset)
+            highlightedImage = UIImage.renderBackgroundImage(fill: secondaryHighlightBackgroundColor, border: secondaryHighlightBorderColor, shadowOffset: Metrics.backgroundShadowOffset)
+        }
 
         setBackgroundImage(normalImage, for: .normal)
-        setBackgroundImage(highlightImage, for: .highlighted)
+        setBackgroundImage(highlightedImage, for: .highlighted)
         setBackgroundImage(disabledImage, for: .disabled)
 
+        activityIndicator.activityIndicatorViewStyle = .gray
+        addSubview(activityIndicator)
+    }
+
+    /// Setup: TitleColor
+    ///
+    private func configureTitleColors() {
+        let titleColorNormal = isPrimary ? primaryTitleColor : secondaryTitleColor
+
         setTitleColor(titleColorNormal, for: .normal)
         setTitleColor(titleColorNormal, for: .highlighted)
-        setTitleColor(titleColorDisabled, for: .disabled)
+        setTitleColor(disabledTitleColor, for: .disabled)
+    }
 
-        activityIndicator.activityIndicatorViewStyle = .gray
+    /// Setup: TitleLabel
+    ///
+    private func configureTitleLabel() {
+        titleLabel?.font = titleFont
+        titleLabel?.adjustsFontForContentSizeCategory = true
+        titleLabel?.textAlignment = .center
+    }
+}
 
-        addSubview(activityIndicator)
+// MARK: - Nested types
+private extension NUXButton {
+    /// Style: Primary
+    ///
+    struct Primary {
+        static let normalBackgroundColor = WPStyleGuide.mediumBlue()
+        static let normalBorderColor = WPStyleGuide.wordPressBlue()
+        static let highlightBackgroundColor = WPStyleGuide.wordPressBlue()
+        static let highlightBorderColor = normalBorderColor
+    }
+
+    /// Style: Secondary
+    ///
+    struct Secondary {
+        static let normalBackgroundColor = UIColor.white
+        static let normalBorderColor = WPStyleGuide.greyLighten20()
+        static let highlightBackgroundColor = WPStyleGuide.greyLighten20()
+        static let highlightBorderColor = highlightBackgroundColor
+    }
+
+    /// Style: Disabled
+    ///
+    struct Disabled {
+        static let backgroundColor = UIColor.white
+        static let borderColor = WPStyleGuide.greyLighten30()
+    }
+
+    /// Style: Title
+    ///
+    struct Title {
+        static let primaryColor = UIColor.white
+        static let secondaryColor = WPStyleGuide.darkGrey()
+        static let disabledColor = WPStyleGuide.greyLighten30()
```

Comment:
Same comment as above: Shouldn't these come from the new Style struct?

Reply:
✔️